xbmc/music/MusicDatabase.cpp

```diff
@@ -1987,12 +2022,14 @@ bool CMusicDatabase::GetTop100Albums(VECALBUMS& albums)
     if (NULL == m_pDB.get()) return false;
     if (NULL == m_pDS.get()) return false;
 
-    // NOTE: The song.idAlbum is needed for the group by, as for some reason group by albumview.idAlbum doesn't work
-    //       consistently - possibly an SQLite bug, as it works fine in SQLiteSpy (v3.3.17)
-    std::string strSQL = "select albumview.* from albumview "
-                    "where albumview.iTimesPlayed>0 and albumview.strAlbum != '' "
-                    "order by albumview.iTimesPlayed desc "
-                    "limit 100 ";
+    // Get data from album and album_artist tables to fully populate albums
+    std::string strSQL = "SELECT albumview.*, albumartistview.* FROM albumview "
+      "LEFT JOIN albumartistview ON albumview.idAlbum = albumartistview.idAlbum "
```

Comment:
Left join because we can have albums without artists, and we still want to see them. It was discussed in another PR, and @Tolriq confirmed that we do have albums and songs without artists sometimes.  An abberation but could happen with tags as data source.  I don't see absence of artist causing an issue here.